curriculum/challenges/english/20-upcoming-python/learn-string-manipulation-by-building-a-cipher/6552487e689f6e895f658717.md

```diff
@@ -0,0 +1,57 @@
+---
+id: 6552487e689f6e895f658717
+title: Step 23
+challengeType: 20
+dashedName: step-23
+---
+
+# --description--
+
+Strings are immutable, which means they cannot be changed once created, for example, by replacing a character with another one.
+This does not mean you cannot re-assign a variable to another string. For example, you could write something like:
+
+```py
+message = 'Hello World'
+message = 'Hello there!'
+```
+
+But the code above does not change the string `Hello World`. Instead, it makes the `text` variable point to a completely new string.
+
+That said, now you need to create a `new_char` variable at the end of your loop body. Set its value to `alphabet[new_index]`.
```

Comment:
The lesson and instructions have nothing to do with one another? Why not make `new_char` -> `char`?

````suggestion
Strings are immutable, which means they cannot be changed once created. However, a variable can be reassigned another string:

```py
message = 'Hello World'
message = 'Hello there!'
```

That said, now you need to create a `new_char` variable at the end of your loop body. Set its value to `alphabet[new_index]`.
````

Reply:
Originally `new_char` was `char`. Then, I realized that it was not working as I thought and I changed only part of the description.